lib/dat-manager.js

```diff
@@ -111,6 +111,25 @@ module.exports = class Manager extends EventEmitter {
     }, cb)
   }
   _activate (dat) {
+    Object.defineProperty(dat, 'progress', {
+      get: () => {
+        const stats = dat.hyperStats && dat.hyperStats.get()
+
+        let progress = 0
+        let total = 0
+
+        if (dat.archive) {
+          progress += dat.archive.metadata.blocks - dat.archive.metadata.blocksRemaining()
+          total += dat.archive.metadata.blocks
+        }
+        if (stats) {
+          progress += stats.blocksProgress
```

Comment:
FYI been seeing a but with hyperdrive stats `blocksProgress` being greater than `blocksTotal`. I have a [workaround in dat-next](https://github.com/joehand/dat-next/blob/master/lib/download.js#L109-L112). That I think solves it. But I'm pretty sure its the same calculation so I have no idea what is going on. 

Sorry haven't had time to chase that down...